### DISCUSSION TITLE
Changes in GraphQL schema

### DISCUSSION THREAD

**Discussion by vikiival** [kodadot/nft-gallery#3929] [Q&A] [answered]
The current GraphQL schema that we are using was made more than one year ago. 

Short after that we introduced resolvers and small schema changes to mitigate the some issues. 

With the upcoming multicurrency on BSX I will personally upgrade the schema with asset. And also remove decimals from the price. 

The question stays the same. Are there any data that you are currently missing?

Cc @kodadot/dev-guild

**Comment by roiLeo** [comment]
> Are there any data that you are currently missing?

For the moment, I see that the following are missing:
- emotes count for rmrk
- some filters & sorting options (Nft mime-type, nsfw, Most Emotes for rmrk)
- sort collection from nft data?
- profil resolver with banner
- lastEvent resolver for moonsama
- collection name in lastEvent resolver
- & stats resolver
- nfts length (totalcount) in collection

Normally you can find issues for most of these points

**Comment by vikiival** [comment]
GraphQL schema will be extended to: 
- Metadata Attributes will be entity so we can implement #3997
- Price will be denominated to the price for frontend (instead of having `1000000000000` you will get `1`)
- each NFT will have `asset` stored
- Chain metadata will be loaded into GraphQL indexer

**Comment by vikiival** [comment]
Also it would be nice to get some inspiration from:
https://twitter.com/reservoir0x/status/1573386810377801728

**Comment by vikiival** [comment, marked as answer]
To @kodadot/internal-dev if you are missing any data for **kodadot/stick** please open an issue in the repository, I will try to accommodate these changes into schema

For general wishes we have the **kodadot/loligo** repository
